Hardware/Hardware/CardReader/CardReaderNetworkAdapter.swift

```diff
@@ -0,0 +1,7 @@
+/// Abstraction provided by Hardware so that clients of this library
+/// can model a way to provide a connection token.
+/// It is meant to abstract an implementation of the [adapter pattern](https://en.wikipedia.org/wiki/Adapter_pattern)
+/// This name is terrible. I can't think of a better one though
+public protocol CardReaderNetworkingAdapter {
+    func fetchToken(completion: @escaping(String?, Error?) -> Void)
```

Comment:
The response returned by `wp-json/wc/v3/payments/connection_tokens` includes not just a token string (in `secret`, but also a `test_mode` boolean. We can just do with the token string for now, but we will probably want to capture that `test_mode` property as well in the future. Maybe open a separate issue for that.

As for a name, CRNA is fine, or perhaps CardReaderConnectionTokenProvider

Reply:
Nevermind `test_mode` I see it below

Reply:
🤔 that's a good point. Maybe we should rename this to something that indicates that it provides "backend configuration parameters"? CardReaderRemoteConfigAdapter? That's too long for my liking, but maybe RemoteConfigAdapter? RemoteConfigProvider?